xmodule/modulestore/xml.py

```diff
@@ -899,26 +917,10 @@ def get_id(org, library, url_name):  # lint-amnesty, pylint: disable=arguments-d
         """
         return LibraryLocator(org=org, library=library)
 
-    @staticmethod
-    def patch_block_kvs(library_block):
-        """
-        Metadata inheritance can be done purely through XBlocks, but in the import phase
-        a root block with an InheritanceKeyValueStore is assumed to be at the top of the hierarchy.
-        This should change in the future, but as XBlocks don't have this KVS, we have to patch it
-        here manually.
-        """
-        init_dict = {key: getattr(library_block, key) for key in library_block.fields.keys()}
-        # if set, invalidate '_unwrapped_field_data' so it will be reset
-        # the next time it will be called
-        lazy.invalidate(library_block, '_unwrapped_field_data')
-        # pylint: disable=protected-access
-        library_block._field_data = inheriting_field_data(InheritanceKeyValueStore(init_dict))
```

Comment:
I know we do some weird things around [library defaults vs. inheritance](https://github.com/openedx/edx-platform/blob/aa68ea116208c430fce2e1aac14ecc50d476da02/xmodule/modulestore/inheritance.py#L383-L386) for library blocks. Is this part definitely safe to remove?

Reply:
This code was a hack to monkey patch the FieldData instance on the root library XBlock to be an `InheritanceKeyValueStore` during import, because the author noticed some errors if that wasn't done. But I found that by making the changes I made to [xmodule/modulestore/inheritance.py](https://github.com/openedx/edx-platform/pull/32273/files/aa68ea116208c430fce2e1aac14ecc50d476da02#diff-174364edda579bd95854e17d32c0c2aa59df72c9474c66ea462762c7bc00b587), this hack is unnecessary. All of the related tests in [xmodule/tests/test_import.py](https://github.com/openedx/edx-platform/pull/32273/files/aa68ea116208c430fce2e1aac14ecc50d476da02#diff-761df6793ec276652eedec1d4b3e04bb622d440f2908a2d100c1d910249f1a3b) which are explicitly testing library inheritance are still passing, and now they don't need the monkey-patching anymore.

In any case, there is no mechanism in Studio to set any attributes on the root `library` XBlock, so I believe that even if this changed behavior, it would only affect users who edited the root XBlock OLX to set some inherited fields into the library, which I don't think is something we encourage or support. In fact, I think the link that you included in your comment shows that field data never comes from the root library block anyways - it either comes from the library component or overrides set onto the component once it's used in a course. 

P.S. I believe the quote "as XBlocks don't have this KVS" is referring to how `parse_xml` gives XModules a different KVS than XBlocks during import, causing different behavior at the time this was written for the `course` XModule as for the `library` XBlock.